src/optimizer/transforms/group-single-chars-to-char-class.js

```diff
@@ -0,0 +1,54 @@
+/**
+ * The MIT License (MIT)
+ * Copyright (c) 2017-present Dmitry Soshnikov <dmitry.soshnikov@gmail.com>
+ */
+
+'use strict';
+
+/**
+ * A regexp-tree plugin to replace single char group disjunction to char group
+ *
+ * (a|b|c) -> ([abc])
+ * (?:a|b|c) -> [abc]
+ */
+module.exports = {
+  Group(path) {
+    const { node } = path;
+
+    const charset = new Map();
+
+    if (!processExpressionCharset(node.expression, charset)
+      || !charset.size
+    ) {
+      return;
+    }
+
+    const characterClass = {
+      type: 'CharacterClass',
+      expressions: Array.from(charset.values())
+    };
+
+    if (node.capturing) {
+      node.expression = characterClass;
+    }
+    else path.replace(characterClass);
+  }
+}
+
+function processExpressionCharset(expression, charset) {
```

Comment:
Let's call it as a predicate, with `should` prefix: `shouldProcessExpressionCharset` (otherwise `processExpressionCharset` sounds like action method).